instaclustr/resource_gcp_vpc_peering.go

```diff
@@ -0,0 +1,166 @@
+package instaclustr
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceGCPVpcPeering() *schema.Resource {
+	return &schema.Resource{
+		Create: GCPresourceVpcPeeringCreate,
+		Read:   GCPresourceVpcPeeringRead,
+		Update: resourceVpcPeeringUpdate,
+		Delete: resourceVpcPeeringDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceVpcPeeringStateImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"vpc_peering_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"cdc_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"peer_vpc_network_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_project_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"peer_subnet": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnets"},
+			},
+
+			"peer_subnets": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:      true,
+				ConflictsWith: []string{"peer_subnet"},
+			},
+		},
+	}
+}
+
+func GCPresourceVpcPeeringCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*Config).Client
+
+	cdcID, err := VpcPeeringCreate(d, meta)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating VPC peering request: %s", err)
+	}
+	var createData CreateGCPVPCPeeringRequest
+	createData, err = GCPcreateVpcPeeringRequest(d)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+
+	var jsonStr []byte
+	jsonStr, err = json.Marshal(createData)
+	if err != nil {
+		return fmt.Errorf("[Error] Error creating GCP VPC peering request: %s", err)
+	}
+	var id string
+	id, err = client.CreateVpcPeering(cdcID, jsonStr)
```

Comment:
**[Maintainability,Required]** It's really hard to see the difference between this func `CreateVpcPeering` and `VpcPeeringCreate` used above. What is the difference? Why does this return an error for cluster creation if it is VPC creation?

Reply:
While `VpcPeeringCreate` has all common data and is being called here the `CreateVpcPeering` is actually creating the peering request after being called from the `api_client.go` file. Thanks for pointing out the error the message here is wrong and should be something like `Failed to call common create functions`